File: src/main/java/com/github/dwendelen/platformd/rest/domain/account/Account.java
package com.github.dwendelen.platformd.rest.domain.account;

import com.datastax.driver.mapping.annotations.ClusteringColumn;
import com.datastax.driver.mapping.annotations.Column;
import com.datastax.driver.mapping.annotations.PartitionKey;
import com.datastax.driver.mapping.annotations.Table;


import java.math.BigDecimal;
import java.util.UUID;

@Table(name = "account")
public class Account {
    @PartitionKey
    @Column(name = "user_id")
    private UUID userId;

    @ClusteringColumn
    @Column(name = "account_id")
    private UUID accountId;

    @Column(name = "account_name")
    private String name;

    @Column(name = "balance")
    private BigDecimal balance;

    public UUID getUserId() {
        return userId;
    }

    public Account setUserId(UUID userId) {
        this.userId = userId;
        return this;
    }

    public UUID getAccountId() {
        return accountId;
    }

    public Account setAccountId(UUID accountId) {
        this.accountId = accountId;
        return this;
    }

    public String getName() {
        return name;
    }

    public Account setName(String name) {
        this.name = name;
        return this;
    }

    public BigDecimal getBalance() {
        return balance;
    }

    public Account setBalance(BigDecimal balance) {
        this.balance = balance;
        return this;
    }
}

File: src/main/scala/com/github/dwendelen/platformd/rest/domain/account/Account.scala
package com.github.dwendelen.platformd.rest.domain.account

import java.time.LocalDate
import java.util.UUID

import com.datastax.driver.mapping.annotations._
import com.fasterxml.jackson.annotation.JsonFormat

import scala.annotation.meta.field
import scala.beans.BeanProperty

@Table(name = "account")
class Account(
                 @(PartitionKey @field)
                 @(Column @field)(name = "user_id")
                 @BeanProperty
                 var userId: UUID,
                 @(ClusteringColumn @field)
                 @(Column @field)(name = "account_id")
                 @BeanProperty
                 var accountId: UUID,
                 @(Column @field)(name = "account_name")
                 @BeanProperty
                 var name: String,
                 @(Column @field)(name = "balance")
                 @BeanProperty
                 var balance: BigDecimal
             ) {
    def this() = this(null, null, null, null)
}

@Table(name = "transaction")
class Transaction(
                     @(PartitionKey @field)
                     @(Column @field)(name = "account_id")
                     @BeanProperty
                     var accountId: UUID,
                     @ClusteringColumn(0)
                     @(Column @field)(name = "transaction_date")
                     @(JsonFormat @field)(shape = JsonFormat.Shape.STRING)
                     @BeanProperty
                     var transactionDate: LocalDate,
                     @ClusteringColumn(1)
                     @(Column @field)(name = "transaction_id")
                     @BeanProperty
                     var transactionId: UUID,
                     @(Column @field)(name = "budget_item")
                     @BeanProperty
                     var budgetItem: UUID,
                     @(Column @field)(name = "amount")
                     @BeanProperty
                     var amount: BigDecimal,
                     @(Column @field)(name = "comment")
                 @BeanProperty
                     var comment: String
                 ) {
    def this() = this(null, null, null, null, null, null)
}
